llvm/lib/Target/AMDGPU/AMDGPUPostLegalizerCombiner.cpp

```diff
@@ -159,17 +160,14 @@ bool AMDGPUPostLegalizerCombinerImpl::tryCombineAll(MachineInstr &MI) const {
 }
 
 bool AMDGPUPostLegalizerCombinerImpl::matchFMinFMaxLegacy(
-    MachineInstr &MI, FMinFMaxLegacyInfo &Info) const {
-  // FIXME: Type predicate on pattern
-  if (MRI.getType(MI.getOperand(0).getReg()) != LLT::scalar(32))
-    return false;
-
-  Register Cond = MI.getOperand(1).getReg();
-  if (!MRI.hasOneNonDBGUse(Cond) ||
-      !mi_match(Cond, MRI,
-                m_GFCmp(m_Pred(Info.Pred), m_Reg(Info.LHS), m_Reg(Info.RHS))))
+    MachineInstr &MI, MachineInstr &FCmp, FMinFMaxLegacyInfo &Info) const {
+  if (!MRI.hasOneNonDBGUse(FCmp.getOperand(0).getReg()))
```

Comment:
Is there a way to do this check in the MIR pattern?

Reply:
No. Other uses could have sneaked in.

Reply:
Sneaked in after the autogenerated matcher runs but before this match function is called? That doesn't sound right.

Reply:
You could:
```
GFCmp *Cmp = cast<GFCmp>(&FCmp);
```

Reply:
The G_FCMP could be used by other MIs.

Reply:
> The G_FCMP could be used by other MIs.

I know. I would like to write something in the pattern to check for this, instead of writing C++ to check for this.

Reply:
Open an issue?